src/asynchro.rs

```diff
@@ -553,7 +563,7 @@ where
         let needed_input_size =
             (chunk_size as f64 / resample_ratio).ceil() as usize + 2 + interpolator.len() / 2;
         let buffer =
-            vec![vec![T::zero(); 11 * needed_input_size + 2 * interpolator.len()]; nbr_channels];
+            vec![vec![T::zero(); (max_resample_ratio_relative as usize + 1) * needed_input_size + 2 * interpolator.len()]; nbr_channels];
```

Comment:
Is this correct? I'm going by what the code was before and how you've changed it in recent commits, but I'm not certain this is correct in all cases.

Reply:
I think this should be correct:
```
((max_resample_ratio_relative + 1.0) * needed_input_size as f64) as usize + 2 * interpolator.len()
```